webapp/src/DOMJudgeBundle/Service/ScoreboardService.php

```diff
@@ -306,6 +311,22 @@ public function calculateScoreRow(Contest $contest, Team $team, Problem $problem
             ->setParameter(':cid', $contest->getCid())
             ->orderBy('s.submittime');
 
+        if ($useExternalJudgements) {
+            $queryBuilder
+                ->addSelect('ej')
+                ->leftJoin('s.external_judgements', 'ej')
+                // We want the external judgement that was started the latest, because that is the
+                // current valid one. We do this by left-joining on external judgements again, bot
```

Comment:
What about marking the external judgements as valid/invalid during import? This would make this query easier and closer to the original.

Reply:
I thought about this, but how are we going to do that properly? The feed could expose runs in an order that differs from sorting by `starttime`. Do we then, every time we encounter a judgement, update all judgements to `valid=false` except the one with the highest `starttime`? This would be kinda similar to what we do here, but then we only do it in one spot.

Reply:
Yes, I think that would be nicer to do it during import. It should be rare anyway, so it's not costly.

Reply:
I will come up with a way to do that and then remove the ugliness here.